fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java

```diff
@@ -98,6 +103,33 @@ public List<OptExpression> transform(OptExpression input, OptimizerContext conte
         predicates = scalarOperatorRewriter.rewrite(predicates,
                 ScalarOperatorRewriter.DEFAULT_REWRITE_SCAN_PREDICATE_RULES);
         predicates = Utils.transTrue2Null(predicates);
+        if (context.getSessionVariable().isEnableScanPredicateExprReuse()) {
+            OlapTablePredicateExtractor olapTablePredicateExtractor =
+                    new OlapTablePredicateExtractor(logicalScanOperator.getColRefToColumnMetaMap());
+            olapTablePredicateExtractor.extract(predicates);
+            ScalarOperator pushedPredicates = olapTablePredicateExtractor.getPushedPredicates();
+            ScalarOperator reservedPredicates = olapTablePredicateExtractor.getReservedPredicates();
+            boolean newScanPredicateIsSame = Objects.equals(pushedPredicates, scanPredicate);
+            boolean newFilterPredicateIsSame = Objects.equals(filterPredicate, reservedPredicates);
+            // if nothing changed after extracting predicates, just return
+            if (newScanPredicateIsSame && newFilterPredicateIsSame) {
+                return Lists.newArrayList();
+            }
+            if (reservedPredicates != null) {
+                Operator.Builder builder = OperatorBuilderFactory.build(logicalScanOperator);
+                LogicalScanOperator newScanOperator = (LogicalScanOperator) builder.withOperator(logicalScanOperator)
+                           .setPredicate(pushedPredicates).build();
```

Comment:
Is this different from original push-down policy? does this can cause some bad cases?

Reply:
> Is this different from original push-down policy? does this can cause some bad cases?

I tested some predicates, and the predicates that are extracted to the filter operator now  were not be pushed down to the storage layer before, so the theoretical behavior is the same as before.